assets/js/components/Dashboard/PermanentConfiguration/Workflows/GlEink/GlEinkWorkflow.tsx

```diff
@@ -1,25 +1,48 @@
-import React, { ComponentType, useState } from "react";
-import { WorkflowProps } from "../../ConfigureScreensPage";
+import React, { ComponentType, useLayoutEffect, useState } from "react";
 import ConfigureScreensWorkflowPage, {
   PlaceIdsAndNewScreens,
 } from "./ConfigureScreensPage";
 import BottomActionBar from "../../BottomActionBar";
-import { useNavigate } from "react-router-dom";
+import { useLocation, useNavigate } from "react-router-dom";
 import StationSelectPage from "./StationSelectPage";
 import { Place } from "../../../../../models/place";
 import { putPendingScreens } from "../../../../../utils/api";
+import { useScreenplayContext } from "../../../../../hooks/useScreenplayContext";
+
+interface EditNavigationState {
+  placeID: string;
+}
+
+const GlEinkWorkflow: ComponentType = () => {
+  const { places } = useScreenplayContext();
+  const location = useLocation();
+  const navigate = useNavigate();
 
-const GlEinkWorkflow: ComponentType<WorkflowProps> = ({
-  places,
-}: WorkflowProps) => {
   const [selectedPlaces, setSelectedPlaces] = useState<Set<string>>(new Set());
   const [configVersion, setConfigVersion] = useState<string>("");
+  const [isEditing, setIsEditing] = useState<boolean>(false);
+  const [configStep, setConfigStep] = useState<number>(0);
 
   const [placesAndScreensToUpdate, setPlacesAndScreensToUpdate] =
     useState<PlaceIdsAndNewScreens>({});
 
-  const navigate = useNavigate();
-  const [configStep, setConfigStep] = useState<number>(0);
+  const getPlacesList = () => {
+    return places.filter((place) =>
+      place.routes.some((route) => route.startsWith("Green"))
+    );
+  };
+
+  useLayoutEffect(() => {
```

Comment:
Pardon my unfamiliarity with React, but why do we need to use `useLayoutEffect` over `useEffect` in this case? My basic understanding is that `useLayoutEffect` makes sure to run synchronously before the browser repaints, but what about the logic requires that exactly?

Reply:
It's actually a way to make the transition smoother when navigating from the Edit button on the Pending page (not present in this PR because task is still in progress). With `useEffect`, the page renders, the hook runs, the component rerenders from the state change. You can actually see the layout change when this happens, which looks odd. Using `useLayoutEffect` makes sure that the state is set correctly before the component renders for the first time.